frappe/share.py

```diff
@@ -13,7 +13,22 @@
 
 
 @frappe.whitelist()
-def add(
+def add(doctype, name, user=None, read=1, write=0, submit=0, share=0, everyone=0, notify=0):
+	"""Expose function without flags to the client-side"""
+	return add_docshare(
```

Comment:
This change wasn't needed because `flags` is explicitly removed from `form_dict` in `frappe.get_newargs`.